src/network/ledger.rs

```diff
@@ -622,20 +620,17 @@ impl<N: Network, E: Environment> Ledger<N, E> {
 
         match self.canon.revert_to_block_height(block_height) {
             Ok(removed_blocks) => {
-                info!("Ledger successfully reverted to block {}", self.canon.latest_block_height());
+                info!(
+                    "Ledger successfully reverted {} blocks down to block {}",
+                    removed_blocks.len(),
+                    self.canon.latest_block_height()
+                );
 
                 // Update the last block update timestamp.
                 *self.last_block_update_timestamp.write().await = Instant::now();
                 // Set the terminator bit to `true` to ensure the miner resets state.
                 self.terminator.store(true, Ordering::SeqCst);
 
-                // Lock unconfirmed_blocks for further processing.
-                let mut unconfirmed_blocks = self.unconfirmed_blocks.write().await;
-
-                // Ensure the removed blocks are not in the unconfirmed blocks.
-                for removed_block in removed_blocks {
-                    unconfirmed_blocks.remove(&removed_block.previous_block_hash());
-                }
```

Comment:
Please keep this logic. While we have minimized the "stuck on wrong fork due to fast-forward with incorrect next block" case (which this logic mitigates), it is better to keep it as it is critical for syncing.